cinema/urls.py

```diff
@@ -1,10 +1,50 @@
-from django.urls import path
+from django.urls import path, include
+from rest_framework import routers
 
-from cinema.views import movie_list, movie_detail
+
+from cinema.views import (
+    MovieViewSet,
+    GenreList,
+    GenreDetail,
+    ActorList,
+    ActorDetail,
+    CinemaHallList,
+    CinemaHallDetail
+)
+
+
+router = routers.DefaultRouter()
+router.register("movies", MovieViewSet, basename="movie-list")
+
+movie_list = MovieViewSet.as_view(
+    actions={"get": "list",
+             "post": "create"}
+)
+
+movie_detail = MovieViewSet.as_view(
+    actions={"get": "retrieve",
+             "put": "update",
+             "patch": "partial_update",
+             "delete": "destroy"}
+)
 
 urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path("genres/", GenreList.as_view(), name="genre-list"),
+    path("genres/<int:pk>/", GenreDetail.as_view(), name="genre-detail"),
+    path("actors/", ActorList.as_view(), name="actor-list"),
+    path("actors/<int:pk>/", ActorDetail.as_view(), name="actor-detail"),
+    path("cinema-halls/", CinemaHallList.as_view(
+        actions={"get": "list", "post": "create"}
+    ), name="cinema-hall-list"),
+    path("cinema-halls/<int:pk>/",
+         CinemaHallDetail.as_view(
+             actions={"get": "retrieve",
+                      "put": "update",
+                      "patch": "partial_update",
+                      "delete": "destroy"}
+         ),
+         name="cinema-hall-detail"),
```

Comment:
The `CinemaHallList` and `CinemaHallDetail` views are being used with `as_view()` and actions, which is typically for viewsets. Ensure these views are indeed viewsets or adjust the implementation to use `ListCreateAPIView` and `RetrieveUpdateDestroyAPIView` if they are class-based views.